### ISSUE TITLE
[Achievement] Hodir - Ulduar

### ISSUE THREAD

**Issue by philoudu16** [WoWManiaUK/Redemption#2230]
The achievements for Hodir in Ulduar are currently unavailable.
These are:
- "I could say that this cache was rare" which is obtained from defeading Hodir under three minutes
- "Getting cold in here" which is obtained for having no raid member getting more than two stacks of the Biting
- "Staying buffed all winter" which is obtained for having all three buffs from the NPCs at the same time
- "Cheese the freeze" which is obtained by defeating Hodir without anyone getting hit by flash freeze
- "I have the coolest friends" which is obtained by defeating Hodir without letting any NPC die during the fight

Link to Hodir:
http://www.wowhead.com/npc=32845/hodir

**Comment by philoudu16** [comment]
We did Ulduar 10 yesterday on Redemption (we got the sigil, but didn't get "I could say that this cache was rare", I was pretty certain to have all buffs at once at some point, we were all carefull about our stacks of debuff, and none of the friendly NPCs died, and none of u got any of the achievements) so I tested them all out on the test server an hour ago (I was alone on dev server though if that can make a difference at all).

**Comment by fvicaria** [comment]
You can only get that in Hard Mode.
Also, that is one of the achievs you listed all.

**Comment by philoudu16** [comment]
We did have a lot of time to spare: we were all in icc gear, knew what we were doing, were using the NPC buffs and we had a shaman for bloodlust. We did it in 1:23 according to dbm (so give or take 10 seconds if the dbm timer doesn't start quite at the right moment).

If you are able to get the sigil, it means you did the hard mode (which is defeating him under three minutes) which means you should get the achievement.

**Comment by fvicaria** [comment]
We will check all of them on dev.

**Comment by Keader** [comment]
Someone can confirm?
If confirmed, i can take a look tomorrow

**Comment by Dranah** [comment]
I know for sure the "staying buffed all winter" isn't doable solo, because you need the "spread" version of the lightning buff, not the original cast on yourself.
I'll test them all anyway.

**Comment by Dranah** [comment]
Confirmed. None of the achievements work.

**Comment by Dranah** [comment]
Re tested just now and none of the achievements work still. Also the hardmode chest contained only 1, non-hardmode level item.
This test was made on 25man.

**Comment by Keader** [comment]
Nothing changed about Hard Mode loot, this issue is about ACHIEVEMENT.
And i dont think that Janitor build it yet.

**Comment by Keader** [comment]
Yes, was not builded yet.
In next reset will be ready to test Dranah.

**Comment by Dranah** [comment]
-.-
put on the fixed label only when its ready for test

**Comment by Dranah** [comment]
UPDATE:
"I could say that this cache was rare" achievement works now, all the rest doesn't

**Comment by Dranah** [comment]
Update:
"Cheese the freeze"
"Getting Cold in here"
and "I have the coolest friends!"
now work as intended.

"Staying buffed all winter" is broken still.

**Comment by fvicaria** [comment]
ok will check again.

**Comment by philoudu16** [comment]
All the achievements are now available.

However, two achievements are not working as intended:
"**Cheese the Freeze**" and "**Getting Cold in Here**" both state "Defeat Hodir without **any raid member** ...". But at the moment those achievements seem to be individual: if one player makes sure he doesnt get more than two stack of Biting Cold then he will get the achievement, even if other people in the raid have 10 stacks. (Those who are not careful will not get the achievement though).

**Comment by fvicaria** [comment]
Well the "Cheese the Freeze" worked for Dranah and no change on it was made.

**Comment by fvicaria** [comment]
I find at least weird the difference so I will ask another test.

**Comment by Dranah** [comment]
All the achievements "work" in the way you can obtain them.

However "Cheese the Freeze" and "Getting Cold In Here" are indeed checking for personal completion instead of raidwide criteria, which makes it a lot easier to obtain than it should be.

**Comment by fvicaria** [comment]
I will sort it by tomorrow morning

**Comment by fvicaria** [comment]
Cheese the Freeze is as-is. I will check the other one.

**Comment by Dranah** [comment]
Yes but it is only checking the player himself.
These achievements should be failed if ANY raid member fails

**Comment by Dranah** [comment]
unlike staying buffed all winter, which is a personal achievement

**Comment by philoudu16** [comment]
I may have been mistaken on what the issue truly is.

When I tested it yesterday, I had two characters. I made sure one would be dodging the flash freeze and not getting more than two stacks of biting cold, and I made sure the other one would get hit by flash freeze and would get more than two stacks of biting cold. Then the first character got all the achievements but the other one didnt get "Cheese the Freeze" or "Getting Cold in Here".

However this time, the first character didn't get those two achievements but the second one did. So I did some more testing, and noticed that between the two lots of achievements, the boss did his emote that he does when the encounter starts:

![wowscrnshot_021718_162539](https://user-images.githubusercontent.com/23452669/36342566-d6210ee2-1400-11e8-96c1-c4ee657152f1.jpg)


So what I think is really happening is that the fight thinks it restarts between the two lots of achievements and ends immediately: 
- Fight finishes for player A
- player A gets his achievements
- fight starts for player B
- fight instantly ends for player B, therefore player B gets all achievements because during that "instantaneous fight", all the players met the requirements for the achievements.

I guess with more people, then the fight will continue"resetting" between the achievements, giving all the achievements to everyone but the first person.
